### DISCUSSION TITLE
Flash Of Unstyled Content (FOUC) when building with webpack

### DISCUSSION THREAD

**Discussion by JensErikChristiansen** [nuxt/nuxt#18653] [Questions]
Hi all!

We get FOUC when 1) we build with webpack (we use a dep `grpc-web` that use CommonJS modules), and 2) we use `<NuxtPage>` in app.vue. If we don't use `<NuxtPage>`, we don't get the FOUC (just for testing, as we need multiple pages in our app).

Inline styles load fine, but any styles from the `<style>` tag are loaded afterwards, causing, a FOUC. This only happens with `builder: 'webpack'` in nuxt.config.ts (`@nuxt/webpack-builder` is installed). I tried `webpack.extractCSS: true` as well, but no joy.

To reproduce:
1. start a new nuxt project
2. in nuxt.config.ts, set `builder: 'webpack'`
3. in app.vue, replace template contents with `<NuxtPage />`
4. create pages/index.vue with the contents:
```
<template>
  <div class="indexWrapper">
    <h1 style="background: blue">hi</h1>
  </div>
</template>

<style scoped>
.indexWrapper {
  background: red;
  height: 100vh;
}
</style>
```

Adding and removing`builder: 'webpack'` will introduce/remove FOUC respectively.

Can anyone help?

Thanks,

Jens

**Comment by adlawan** [comment]
Strategy and compatetibe

**Reply by JensErikChristiansen** [reply]
pardon?

**Comment by JensErikChristiansen** [comment]
No solution on this?

**Reply by adlawan** [reply]
Hi can I give you quote

**Comment by wang-qingxuan** [comment]
hi. I'm facing the same problem with nuxt3 & vue3 & webpack5. Did you find any solutions?
